Various changes merged from 17.2 & Change as per Ticket 38425

labkey-bpatel opened the pull request.

gottfred committed:
column title changed to be more clear

gottfred committed:
merge develop into fb_breeding

gottfred committed:
default snomed code to p5-bb320 (diagnostic ultrasound of abdomen and retroperitoneum) when entering an ultrasound

gottfred committed:
added pregnancies to animal abstract

gottfred committed:
support for showing pregnancies for multiple animals added

dnicolalde committed:
Enhancements on Viral Load display.

csebranek committed:
Add expiration date of 3 years to protocol view; add basic species pivot query

dnicolalde committed:
Update protocolExpirationDate.sql

Do not need the group by statement.

dnicolalde committed:
Merge pull request #55 from WNPRC-EHR-Services/develop

Add expiration date of 3 years to protocol view; add basic species pi…

csebranek committed:
Add in counts max animals per species to protocol table

csebranek committed:
Merge branch 'develop' of https://github.com/WNPRC-EHR-Services/wnprc-modules into develop

csebranek committed:
Change label for max animals per species

dnicolalde committed:
Merge pull request #57 from WNPRC-EHR-Services/develop

Add extensible pivot columns to protocol table to view max animals count per species

csebranek committed:
Adding protocols grid view for max animals per species

csebranek committed:
Initial DB configuration for animal requests; remove typescript tsconfig check in javascript gradle file

gottfred committed:
files added for pregnancy notifications and for automatically calculating due dates when in the form view

dnicolalde committed:
Adding R24 scripts for SouthWest project.

dnicolalde committed:
Changing name of clinical encounters to enocunters.

dnicolalde committed:
Merge branch 'develop' of https://github.com/WNPRC-EHR-Services/wnprc-modules into develop

dnicolalde committed:
Enhancements on Viral Load display.

gottfred committed:
added insert and update methods and refactored upsert method to remove duplicated code

gottfred committed:
fixed typo

gottfred committed:
changed pregnancy linking dropdowns to refer to date_conception rather than date_conception_early

gottfred committed:
added method to kick off pregnancy notifications and a method to escalate security to update already completed breeding encounter records

gottfred committed:
fix for possible null pointer

gottfred committed:
completely rewrote code that creates breeding encounters from housing changes to account for many more use cases that I was previously unaware of. also moved all of the logic from js to java

gottfred committed:
cleanup

gottfred committed:
added plugins for calculating due date based on conception date

gottfred committed:
automatically calculate gestation day and pregnancy due dates in both grid and form editors

gottfred committed:
moved update of breeding outcome to java so that security can be escalated

gottfred committed:
beginning of pregnancy notification implementation

gottfred committed:
attempt to fix compilation issues

gottfred committed:
fixed trigger script exception

gottfred committed:
automatically populate room field for observations per cage section on irregular obs page

gottfred committed:
Merge branch 'develop' of https://github.com/WNPRC-EHR-Services/wnprc-modules into develop

dnicolalde committed:
Merge pull request #58 from WNPRC-EHR-Services/develop

Adding protocols grid view for max animals per species, also adding the dam to necropsy, and max animals per protocol

csebranek committed:
modify wnprc_ehr to serve up animalrequests module

csebranek committed:
update animal requests - capitalize select, change recipient email submission

csebranek committed:
Merge branch 'develop' into fb_animalrequests

csebranek committed:
add more fields, use subscribe mode for react final form

gottfred committed:
renamed to actual dataset name rather than its label

gottfred committed:
default boolean values to false if they are not checked

gottfred committed:
only check if assistant and performedby users are current users and/or pathologists if it is a new record, or if the fields have changed

gottfred committed:
changed LABKEY_TEAMCITY_CONFIG and LABKEY_VERSION to arguments that can be passed in, and added a LABKEY_IS_COMMUNITY argument so that a single Dockerfile can be used across all branches/versions

dnicolalde committed:
Merge pull request #59 from WNPRC-EHR-Services/develop

Necropsy Security update

dnicolalde committed:
Enhancements on Viral Load display, and fixing viral challenge query.

gottfred committed:
moved updating old ultrasound records to java and used security escalator so that permissions won't be an issue

gottfred committed:
cleaned up imports

gottfred committed:
only show breeding encounters that occurred within the last 6 months

gottfred committed:
cleanup

Commit:
Change to demographic ViralLoad query.

gottfred committed:
changed which fields are used to calculate gestation day, and changed primary key on gestational_days table

gottfred committed:
move gestation day calculation to trigger script

gottfred committed:
get correct field name

gottfred committed:
updated gestation day calculations to be more robust and to only calculate if within a certain threshold

gottfred committed:
set field to null if there's not a close enough value in the db

gottfred committed:
Merge pull request #60 from WNPRC-EHR-Services/develop

Changes to Viral load and fixes to demographics merge.

gottfred committed:
mark rows as completed upon data insertion

gottfred committed:
merged develop into fb_breeding

gottfred committed:
removed debugging statements

gottfred committed:
removed some console.log debug statements and updated qcstates

gottfred committed:
removed logger

gottfred committed:
added new columns to grid display

gottfred committed:
new and renamed files

gottfred committed:
show only current birth types

gottfred committed:
show only current birth types

gottfred committed:
show birth type description rather than code

gottfred committed:
fix which fields are required

csebranek committed:
Add new fields to form

gottfred committed:
move query to correct schema folder

gottfred committed:
change schemas from study to wnprc

Commit:
Merge branch 'develop' of https://github.com/WNPRC-EHR-Services/wnprc-modules into develop

gottfred committed:
Update Dockerfile

removed default value from LABKEY_IS_COMMUNITY argument to make things more clear

dnicolalde committed:
Merge pull request #61 from WNPRC-EHR-Services/fb_breeding

Merge breeding back into develop

dnicolalde committed:
moving certifcate from secondary mginx instance..

csebranek committed:
restrict active projects to projects with a protocol; add tooltip for infectious disease; fix title of form in manageRecord.view

csebranek committed:
remove clientside email; add back in error for required fields

dnicolalde committed:
Merge pull request #62 from WNPRC-EHR-Services/develop

Move breeding changes to production

Commit:
Adding CADVISOR_PORT and changing DANGERPORT for second instance.

csebranek committed:
Add animal requests module; remove old app location; add server side email trigger

csebranek committed:
Add readme

csebranek committed:
Changing build script name to attempt to make teamcity gradle build happy

csebranek committed:
Another attempt to try to fix teamcity issue

dnicolalde committed:
Adding ignore to cert and key.pem files

dnicolalde committed:
Merge branch 'develop' of github.com:WNPRC-EHR-Services/wnprc-modules into develop

gottfred committed:
show breeding encounters that are ongoing when entering pregnancies and fix for housing changes to not check current housing status

dnicolalde committed:
Merge pull request #63 from WNPRC-EHR-Services/fb_breeding

Breeding fixes

csebranek committed:
Fix css issues with links and menus - remove bootstrap conflicts; send user to main data entry

csebranek committed:
Merging fb_animalrequests into develop; resolved conflict with WNPRC_EHR Module and TriggerScriptHelper

dnicolalde committed:
Merge pull request #65 from WNPRC-EHR-Services/develop

Breeding fixes

csebranek committed:
AnimalRequests: add more tooltips and a note on the form

gottfred committed:
enhanced validations and split multiple sireids into comma separated list

gottfred committed:
customize the sireid column on pregnancies table to link to multiple animals if necessary

gottfred committed:
Merge branch 'develop' into fb_breeding

gottfred committed:
Merge branch 'develop' into fb_breeding

csebranek committed:
no need to register the triggers, no longer using wnprc_triggers.js

gottfred committed:
remove test code

gottfred committed:
Merge branch 'develop' into fb_breeding

gottfred committed:
changed insert to upsert

dnicolalde committed:
Merge pull request #67 from WNPRC-EHR-Services/develop

Breeding fixes and AnimalRequests updates

dnicolalde committed:
Adding information about new certificates.

gottfred committed:
register single form overrides for breeding forms so that javascript dependencies are loaded when editing records

gottfred committed:
added some error checking and error messages

gottfred committed:
refactored to save breeding encounters in onUpsert instead of onComplete so that error messages can be shown

gottfred committed:
changed gestation period for cynos to 165 days

gottfred committed:
add estimated due date and non-range current gestation

gottfred committed:
removed restriction on linking to breeding encounters that already resulted in a pregnancy

gottfred committed:
changed wording of error message

gottfred committed:
added space after comma when displaying multiple sires per user request

dnicolalde committed:
Merge pull request #68 from WNPRC-EHR-Services/develop

Breeding fixes

gottfred committed:
change which column is shown for conception date in outcomes and ultrasounds views

dnicolalde committed:
Adding information about default certificates.

dnicolalde committed:
Merge branch 'develop' of github.com:WNPRC-EHR-Services/wnprc-modules into develop

gottfred committed:
check if the row is being inserted and only create/update a breeding encounter on an insert event

gottfred committed:
Merge branch 'develop' of https://github.com/WNPRC-EHR-Services/wnprc-modules into develop

gottfred committed:
add description field to ultrasounds

gottfred committed:
Added type to ultrasound description

gottfred committed:
change all dates to yyyy-MM-dd format and make gestation day sortable by removing the 'days' text

gottfred committed:
changed formatting of pregnancyidfield text

gottfred committed:
fix issue with remarks not being correctly populated when there are no remarks

dnicolalde committed:
Merge pull request #69 from WNPRC-EHR-Services/develop

change which column is shown for conception date in outcomes and ultr…

gottfred committed:
remove unnecessary code

gottfred committed:
remove 'SUBMIT FINAL' button from breeding related forms

gottfred committed:
add null checks to make sure javascript errors don't occur

csebranek committed:
Only allow numbers for number of animals input field; add Macaque as a species in ehr lookups

csebranek committed:
Add title parameter in email for animal request notification

gottfred committed:
removed console output used for debugging

dnicolalde committed:
Merge pull request #71 from WNPRC-EHR-Services/develop

Breeding: removing submit final button and other small fixes; AnimalRequests: force number input for form and add Macaque as species

gottfred committed:
Allow for pooling multiple samples in a single well

gottfred committed:
show +30, +60, etc even when there are no early/late conception dates

gottfred committed:
Merge branch 'fb_breeding' into develop

gottfred committed:
fix multiple sire display issue in excel/tsv exports

gottfred committed:
fix floating point display issue by truncating to at most 2 digits

gottfred committed:
Merge branch 'fb_breeding' into develop

gottfred committed:
remove 'natural' sort of subject ids to vastly improve load times

gottfred committed:
changed reason to a multi-select combobox and fixed up some field show/hide logic to account for this

gottfred committed:
generate an error message if a user selects both 'Breeding' and 'Breeding ended' for Reason For Move

gottfred committed:
Iterate through all of the reasons that were selected in the Reason For Move field to determine if 'Breeding' or 'Breeding ended' was selected

gottfred committed:
parse and link to each Reason For Move when displaying the dataset if there are multiple reasons

gottfred committed:
Merge branch 'fb_breeding' into develop

gottfred committed:
formatting fixes

gottfred committed:
Merge branch 'fb_breeding' into develop

csebranek committed:
Fix numeric vs string issues in the access report uploader

Commit:
ETL for Chemistry Analyzer

dnicolalde committed:
Merge pull request #72 from WNPRC-EHR-Services/develop

Breeding fixes/enhancements and Virology updates

gottfred committed:
allow for correct sorting/filtering of housing change reasons while still showing the title field when creating a housing change request

gottfred committed:
Merge branch 'fb_breeding' into develop

csebranek committed:
Upgrade R and use the visPedigree R module for more pedigree plots

gottfred committed:
fix sorting/filtering of sireids

gottfred committed:
Merge branch 'fb_breeding' into develop

csebranek committed:
Revert "Upgrade R and use the visPedigree R module for more pedigree plots" breaks kinship2

This reverts commit 0986855646d5ccfb22262cf7553b48016696979c.

dnicolalde committed:
Merge pull request #74 from WNPRC-EHR-Services/develop

Fix sorting or customized columns and split ids for viral samples in a cleaner way

gottfred committed:
add Dockerfile build-arg to set community version flag to true

gottfred committed:
Merge branch 'develop' of https://github.com/WNPRC-EHR-Services/wnprc-modules into develop

gottfred committed:
simplified build arguments

gottfred committed:
fixed labkey Dockerfile to work with 17.2 Community and Premium, as well as 18.3

gottfred committed:
removed unnecessary comments

gottfred committed:
removed unnecessary comment

gottfred committed:
changed LABKEY_IS_COMMUNITY flag to LABKEY_IS_PREMIUM and set default to build for 17.2

gottfred committed:
fixed wording in comment

gottfred committed:
Update README.md

gottfred committed:
Update README.md

gottfred committed:
Update README.md

gottfred committed:
removed unnecessary argument

csebranek committed:
Use R-3.5.1 and install texinfo-tex dependency

Commit:
ChemistryAnalyzer adding comments to results

gottfred committed:
Merge branch 'develop' of https://github.com/WNPRC-EHR-Services/wnprc-modules into develop

gottfred committed:
get the store by name rather than traversing the ext components hierarchy

gottfred committed:
show title rather than value for housing change reason

Commit:
ChemistryAnalyzer importing from Vet Axcel and adding to clinpath request.

gottfred committed:
reorganized files to correct location by removing java folder

gottfred committed:
reorganized files to correct location by removing java folder

csebranek committed:
after merge w master

Commit:
Adding performing_lab to import.

csebranek committed:
AnimalRequests: remove node_modules folder as well when cleaning

Commit:
Adding mySQL to Docker

Addding configuration to connect to mySQL on docker

Commit:
Merge branch 'develop' into fb_chemistryAnalyzer

Commit:
Temporary solution bypassing install.r

Commit:
Installing R dependencies, needs newer version of serverjre to work.

dnicolalde committed:
Merge pull request #75 from WNPRC-EHR-Services/develop

Improvements to the labkey Dockerfile and updates to the tomcat Dockerfile for R pedigree plots
Moving Gringotts src folder.

Commit:
Using alternative identifier to manage upload of chemistry between ETL process, also forcing to always have order_date in mergingPatientsOreders.

Commit:
Only importing data after June 08, 2019 and moving commentText to remark.

csebranek committed:
Fix deprecated method

csebranek committed:
Merge branch 'wnprc18.3_master' of https://github.com/WNPRC-EHR-Services/wnprc-modules into wnprc18.3_master

Commit:
Merge branch 'master' into develop

gottfred committed:
Delete datasets_manifest.xml

gottfred committed:
Delete datasets_metadata.xml

gottfred committed:
Delete snprc_r24.dataset

gottfred committed:
Delete study.xml

gottfred committed:
Delete snprc_r24.xml

gottfred committed:
Delete docker-compose.yml

gottfred committed:
Revert "Delete docker-compose.yml"

This reverts commit 189f207cae130937dbecc6b2a4940f866f701a7b.

gottfred committed:
fixed changes to have mysql info

dnicolalde committed:
Merge pull request #79 from WNPRC-EHR-Services/develop

Update to import data from chemistry analyzer.

gottfred committed:
formatting

gottfred committed:
changed wording from pooled to batched since that's how Virology has been referencing these samples

csebranek committed:
Fix barrier access upload issues

csebranek committed:
Fix pedigree union all issue

csebranek committed:
Merge branch 'wnprc18.3_master' of https://github.com/WNPRC-EHR-Services/wnprc-modules into wnprc18.3_master

dnicolalde committed:
LinkForCompliance

dnicolalde committed:
Change to clinical encounters to encounters

dnicolalde committed:
Merge pull request #83 from WNPRC-EHR-Services/develop

Link to Compliance Occ Health and Safety folder

csebranek committed:
Add new page for linking persons to unidentified cards

csebranek committed:
Add notes to person data panel

csebranek committed:
Some refactoring and added mark as exempt button functionality

csebranek committed:
Use only one json object in api action

csebranek committed:
Add new manner of death lookup

csebranek committed:
Add new manner of death lookup

csebranek committed:
Merge branch 'develop' of https://github.com/WNPRC-EHR-Services/wnprc-modules into develop

csebranek committed:
Revert "Add new manner of death lookup"

This reverts commit f19666aee36dcbe1a39f1236648be642001cfd08.

csebranek committed:
Add new manner of death, forgot sql file before

csebranek committed:
move linking cards into original unidentified cards page

csebranek committed:
Merge branch 'wnprc18.3_master' of https://github.com/WNPRC-EHR-Services/wnprc-modules into wnprc18.3_master

csebranek committed:
Add back bootstrap datetimepicker plugin for TB dashboard pages

dnicolalde committed:
Changes to docker-compose to not run perlscripts by default.

csebranek committed:
Merge pull request #85 from WNPRC-EHR-Services/fb_complianceUpdates

Fb compliance updates for linking unidentified cards to persons

csebranek committed:
Merge remote-tracking branch 'origin/master' into develop

dnicolalde committed:
Merge pull request #86 from WNPRC-EHR-Services/develop

Compliance updates to link persons to cards; and add a new manner of death for necropsies; changes to docker compose for cron backup

csebranek committed:
Specify json format for Date deserialization

csebranek committed:
hacky fix to hide loading bar on no results since table is not an observable

csebranek committed:
add comment for no loading bar fix

dnicolalde committed:
Merge pull request #88 from WNPRC-EHR-Services/develop

Fix to hide loading bar on no results since table is not an obs…

dnicolalde committed:
Changes to major surgery query.

dnicolalde committed:
Changes to food derpive to allow changes to assignedTo field.

dnicolalde committed:
ERROR message - Changes to major surgery query.

dnicolalde committed:
Merge remote-tracking branch 'origin/develop' into develop

dnicolalde committed:
Merge remote-tracking branch 'origin/develop' into develop

dnicolalde committed:
Merge remote-tracking branch 'origin/develop' into develop

gottfred committed:
Merge pull request #89 from WNPRC-EHR-Services/develop

Changes to Food deprive to allow changes assignedTo

csebranek committed:
add json annotation for date field to fix deserialization

csebranek committed:
Explicitly set wnrpc units menu width

csebranek committed:
Add gringotts back in

csebranek committed:
Another sync with master branch

csebranek committed:
Yet another merge with master branch - tb compliance updates, food deprive up, necropsy up

labkey-jeckels committed:
Ticket 38337: Move extjs3 code from SVN to WNPRC github repo

labkey-jeckels committed:
Fix syntax to restoring full config after testing

labkey-jeckels committed:
Ensure JS dependency ordering to avoid problems resolving Ext3

Rename to match dataset name instead of label for obs vs Irregular Observations

labkey-bpatel committed:
Support Ticket 38425: Remove 'Import Historical Charges' link, and update 'charges and adjustments not yet billed' link

labkey-bpatel committed:
Merge remote-tracking branch 'origin/wnprc18.3_master' into wnprc18.3_master

labkey-bpatel committed:
Merge branch 'release19.2-SNAPSHOT' into 18.3w_fb_mergeTo19.2

labkey-jeckels committed:
Update scripts to work during bootstrap after the EHR module added Container columns to ehr_lookups tables. Since production has already run these scripts, we can assume they have either run already, or they'd be running after the EHR module has done its upgrade

labkey-jeckels committed:
Merge remote-tracking branch 'origin/wnprc18.3_master' into wnprc18.3_master

labkey-bpatel commented:
@labkey-jeckels this is a new/renamed branch to test against 19.2 on teamcity - renaming got rid of the PR on an old branch, so had to create this new PR.

labkey-jeckels commented:
Build is running now:

https://teamcity.labkey.org/viewLog.html?buildId=813829&buildTypeId=LabKey_192ReleaseSnapshot_Premium_Ehr_BuildEhr

labkey-jeckels commented:
Build failed:

mnt/teamcity/agent/work/2592c7ee65830c87/externalModules/wnprc-modules/WNPRC_EHR/src/org/labkey/wnprc_ehr/WNPRC_EHRTestController.java:4: error: cannot find symbol
import org.labkey.api.action.ApiAction;
                            ^
  symbol:   class ApiAction
  location: package org.labkey.api.action
/mnt/teamcity/agent/work/2592c7ee65830c87/externalModules/wnprc-modules/WNPRC_EHR/src/org/labkey/wnprc_ehr/WNPRC_EHRTestController.java:38: error: cannot find symbol
    public static class ImportDatasetMetadataAction extends ApiAction<Void>
                                                            ^
  symbol:   class ApiAction
  location: class WNPRC_EHRTestController
/mnt/teamcity/agent/work/2592c7ee65830c87/externalModules/wnprc-modules/WNPRC_EHR/src/org/labkey/wnprc_ehr/WNPRC_EHRTestController.java:101: error: cannot find symbol
    public class CreatePregnanciesAction extends ApiAction<Void>
                                                 ^
  symbol:   class ApiAction
  location: class WNPRC_EHRTestController
/mnt/teamcity/agent/work/2592c7ee65830c87/externalModules/wnprc-modules/WNPRC_EHR/src/org/labkey/wnprc_ehr/WNPRC_EHRController.java:1250: error: cannot find symbol
    @RequiresPermission(AdminPermission.class)
...

labkey-bpatel committed:
Minor updates to be compatible with 19.2. Add LDK dependency for Gringotts module.

labkey-bpatel committed:
Merge branch 'wnprc18.3_master' into 19.2_fb_mergeFrom18.3

labkey-bpatel committed:
Reset module version (from 15.19 back to 18.31). Automated test fix.

labkey-bpatel committed:
Support Ticket 38497: Invoice A/R report - Payment Received on auto generating.
Don't store today's date if pmt received date is not provided, instead validate if pmt amt received is provided then pmt received date is expected.

labkey-bpatel committed:
Moved sortStore() from ehr's utils.js.

labkey-bpatel committed:
Add missing paren.

labkey-bpatel committed:
Update automated test for viewChargesAdjustmentsNotYetBilled -- remove chargeId, since its a rowid and could vary.

labkey-bpatel committed:
Merge remote-tracking branch 'origin/wnprc18.3_master' into 19.2_fb_mergeFrom18.3
